Generals/Code/GameEngine/Source/GameLogic/System/GameLogic.cpp

```diff
@@ -3334,6 +3334,8 @@ void GameLogic::removeObjectFromLookupTable( Object *obj )
 	if( obj == NULL )
 		return;
 
+	DEBUG_ASSERTCRASH( m_objHash.contains(obj->getID()) , ("bad ObjectID: %u when removing object from lookup table, object not found.", (UnsignedInt)obj->getID()) );
```

Comment:
I think it is better to add this as part of the fixing pull request, and use this change only to sync Zero Hour with Generals.